File: T1.SqlSharp/T1.SqlSharp/Expressions/SqlTableHintIndex.cs
namespace T1.SqlSharp.Expressions;

public class SqlTableHintIndex : ISqlExpression
{
    public SqlType SqlType { get; } = SqlType.TableHintIndex;
    public TextSpan Span { get; set; } = new();
    public void Accept(SqlVisitor visitor)
    {
        visitor.Visit_TableHintIndex(this);
    }

    public List<ISqlExpression> IndexValues { get; set; } = [];
    public string ToSql()
    {
        return $"INDEX ({string.Join(", ", IndexValues.Select(x => x.ToSql()))})";
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlTableHintIndex.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlTableHintIndex implements ISqlExpression {
    SqlType: SqlType = SqlType.TableHintIndex;
    Span: TextSpan = new TextSpan();
    IndexValues: ISqlExpression[] = [];

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_TableHintIndex(this);
    }

    ToSql(): string {
        return this.IndexValues.map(x => x.ToSql()).join(', ');
    }
} 